apps/prairielearn/src/lib/question-render.ts

```diff
@@ -468,89 +494,98 @@ export async function getAndRenderVariant(
     group_config,
     authz_result,
   });
-  Object.assign(locals, newLocals);
   if (locals.manualGradingInterface && question?.show_correct_answer) {
-    locals.showTrueAnswer = true;
+    newLocals.showTrueAnswer = true;
   }
+  Object.assign(locals, newLocals);
```

Comment:
Nit: any reason why you assign to locals instead of resultLocals? I know the result is the same, but given resultLocals is intended to represent the values assigned as a result of this function it seems to make sense to assign them there instead. Won't block on this.

Reply:
My rationale is that we end up passing around `locals` to other functions here, and so I want to make it slightly clearer that these values will be used by other called functions. `resultLocals` would be only for things that are read by callers of this function. Does that rationale make sense? Do you see a better way to do all this?

Reply:
This makes sense. I thought of suggesting type predicates, but it seems to be more complex than needed in this case, particularly since you're raising the option of replacing this with return values instead.